homeworks/irina.kaparushkina_IrinKapa/homework_6/editor-1.js

```diff
@@ -0,0 +1,36 @@
+function addName(keyName, obj) {
+    let protectedVal;
+    Object.defineProperty(obj, keyName, {
+        get() {
+            return protectedVal;
+        },
+        set(value) {
+            const array = value.trim().split(' ').map(elem => elem.toLowerCase()).map(elem => elem[0].toUpperCase() + elem.slice(1));
+            protectedVal = array.join(' ').split('-').map(elem => elem[0].toUpperCase() + elem.slice(1)).join('-');
```

Comment:
this looks like something that could be optimized with some kind of `capitalizeFirstChar` function, but FYI this is a good case for RegExp